FLOE-112

jobara opened the pull request:
This is the start of the simple editor to be used by the Metadata demo. While this doesn't include all of the functionality needed for the first iteration of the demo, it does provide a start and something useable for a first nightly build. It also includes many of the dependencies that we'll need going forward. Another pull request will be made to add in other functionality, like inserting media. 

Note that this pull request also includes changes related to FLOE-108.

http://issues.fluidproject.org/browse/FLOE-112

jobara committed:
FLOE-112: simple unstyled editor

jobara committed:
Merge remote-tracking branch 'cindy/FLOE-108' into FLOE-112

jobara committed:
FLOE-112: improved styling and active state

Added bootstrap for styling. Also updated the code to manage the state of the controls.

jobara committed:
FLOE-112: fixed keyup binding

Corrected event name and restricted the handler to only be called when the arrow keys have been pressed.

jobara committed:
FLOE-112: Added persistence and markup viewer

Using pouchDB to add a persistence layer. This also allows for the stored markup to viewed on a separate page.

jobara committed:
FLUID-112: dataSource fires afterChange event

The dataSource now hooks into the onChange event from pouchDB to fire the afterChange event. This will allow live following changes as the happen, and makes it possible for live updating of the markup viewer.

jobara committed:
FLOE-112: Pretty printing markup viewer

jobara committed:
FLOE-112: Pulling in highlight.js dependency

jobara committed:
FLOE-122: Added the pretty-diff dependency

jobara committed:
FLOE-112: added version info for highlight.js

jobara committed:
FLOE-112: removed unneeded line in release notes

jobara committed:
FLOE-112: Added Bootstrap dependency

jobara committed:
FLOE-112: reformating releaseNotes

michelled commented on demos/html/metadata.html:
```diff
@@ -0,0 +1,36 @@
+            <div class="flc-simpleEditor panel panel-default">
+                <ul class="flc-simpleEditor-toolbar btn-toolbar panel-heading" role="toolbar">
+                    <ul class="btn-group">
+                        <li id="bold" class="flc-simpleEditor-control btn btn-default" role="button" data-control="bold">B</li>
```
Any reason for the id here?

jobara committed:
FLOE-112: updated demo markup, added restart

Updated the demo markup, adding in a rough header with restart functionality and a link to the metadata markup view.

Fixed the reset implementation for the simpleEditor, which wasn't saving the changes.

michelled commented:
Merged into project repo at d7409eab6dbb0a1e2bd54f1b37d1fd14c2c1410f
